mailpoet/lib/Subscription/Captcha/Validator/RecaptchaValidator.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace MailPoet\Subscription\Captcha\Validator;
+
+use MailPoet\Settings\SettingsController;
+use MailPoet\Subscription\Captcha\CaptchaConstants;
+use MailPoet\WP\Functions as WPFunctions;
+
+class RecaptchaValidator implements CaptchaValidator {
+
+
+  /** @var SettingsController  */
+  private $settings;
+
+  /** @var WPFunctions  */
+  private $wp;
+
+  public function __construct(
+    SettingsController $settings,
+    WPFunctions $wp
+  ) {
+    $this->settings = $settings;
+    $this->wp = $wp;
+  }
+
+  public function validate(array $data): bool {
+
+    $captchaSettings = $this->settings->get('captcha');
+    if (empty($data['recaptchaResponseToken'])) {
+      throw new ValidationError(__('Please check the CAPTCHA.', 'mailpoet'));
+    }
+
+    $secretToken = $captchaSettings['type'] === CaptchaConstants::TYPE_RECAPTCHA_INVISIBLE ? $captchaSettings['recaptcha_invisible_secret_token'] : $captchaSettings['recaptcha_secret_token'];
+
+    $response = $this->wp->wpRemotePost('https://www.google.com/recaptcha/api/siteverify', [
```

Comment:
I'd suggest defining the URL in a const, what do you think?

Reply:
Addressed in [d86d690](https://github.com/mailpoet/mailpoet/pull/4196/commits/d86d690e9acfc68f90f660f1e67063078625578b)